Mathlib/CategoryTheory/Monoidal/Braided.lean

```diff
@@ -422,286 +490,124 @@ theorem tensor_μ_natural {X₁ X₂ Y₁ Y₂ U₁ U₂ V₁ V₂ : C} (f₁ :
-    by pure_coherence
-  slice_lhs 2 6 => rw [this]
-  clear this
-  slice_lhs 1 3 => rw [← tensor_comp, ← tensor_comp, tensor_μ_def₁, tensor_comp, tensor_comp]
-  slice_lhs 3 4 => rw [← tensor_id, associator_inv_naturality]
-  slice_lhs 4 5 => rw [← tensor_comp, associator_naturality, tensor_comp]
-  slice_lhs 5 6 =>
-    rw [← tensor_comp, ← tensor_comp, associator_naturality, tensor_comp, tensor_comp]
-  slice_lhs 6 10 =>
-    rw [← tensor_comp, ← tensor_comp, ← tensor_comp, ← tensor_comp, ← tensor_comp, ← tensor_comp, ←
-      tensor_comp, ← tensor_comp, tensor_id, tensor_associativity_aux, ← tensor_id, ←
-      id_comp (𝟙 X₁ ≫ 𝟙 X₁ ≫ 𝟙 X₁ ≫ 𝟙 X₁ ≫ 𝟙 X₁), ← id_comp (𝟙 Z₂ ≫ 𝟙 Z₂ ≫ 𝟙 Z₂ ≫ 𝟙 Z₂ ≫ 𝟙 Z₂),
-      tensor_comp, tensor_comp, tensor_comp, tensor_comp, tensor_comp, tensor_comp, tensor_comp,
-      tensor_comp, tensor_comp, tensor_comp]
-  slice_lhs 11 12 =>
-    rw [← tensor_comp, ← tensor_comp, Iso.hom_inv_id]
-    simp
-  simp only [assoc, id_comp]
-  slice_lhs 10 11 =>
-    rw [← tensor_comp, ← tensor_comp, ← tensor_comp, Iso.hom_inv_id]
-    simp
-  simp only [assoc, id_comp]
-  slice_lhs 9 10 => rw [associator_naturality]
-  slice_lhs 10 11 => rw [← tensor_comp, associator_naturality, tensor_comp]
-  slice_lhs 11 13 => rw [tensor_id, ← tensor_μ_def₂]
-  have :
-    ((𝟙 X₁ ⊗ (α_ (X₂ ⊗ Y₁) Z₁ Y₂).inv) ⊗ 𝟙 Z₂) ≫
-        ((𝟙 X₁ ⊗ (α_ X₂ Y₁ Z₁).hom ⊗ 𝟙 Y₂) ⊗ 𝟙 Z₂) ≫
-          (α_ X₁ ((X₂ ⊗ Y₁ ⊗ Z₁) ⊗ Y₂) Z₂).hom ≫
-            (𝟙 X₁ ⊗ (α_ (X₂ ⊗ Y₁ ⊗ Z₁) Y₂ Z₂).hom) ≫
-              (𝟙 X₁ ⊗ (α_ X₂ (Y₁ ⊗ Z₁) (Y₂ ⊗ Z₂)).hom) ≫ (α_ X₁ X₂ ((Y₁ ⊗ Z₁) ⊗ Y₂ ⊗ Z₂)).inv =
-      (α_ X₁ ((X₂ ⊗ Y₁) ⊗ Z₁ ⊗ Y₂) Z₂).hom ≫
-        (𝟙 X₁ ⊗ (α_ (X₂ ⊗ Y₁) (Z₁ ⊗ Y₂) Z₂).hom) ≫
-          (𝟙 X₁ ⊗ (α_ X₂ Y₁ ((Z₁ ⊗ Y₂) ⊗ Z₂)).hom) ≫
-            (α_ X₁ X₂ (Y₁ ⊗ (Z₁ ⊗ Y₂) ⊗ Z₂)).inv ≫
-              (𝟙 (X₁ ⊗ X₂) ⊗ 𝟙 Y₁ ⊗ (α_ Z₁ Y₂ Z₂).hom) ≫ (𝟙 (X₁ ⊗ X₂) ⊗ (α_ Y₁ Z₁ (Y₂ ⊗ Z₂)).inv) :=
-    by pure_coherence
-  slice_lhs 7 12 => rw [this]
-  clear this
-  slice_lhs 6 7 => rw [associator_naturality]
-  slice_lhs 7 8 => rw [← tensor_comp, associator_naturality, tensor_comp]
-  slice_lhs 8 9 => rw [← tensor_comp, associator_naturality, tensor_comp]
-  slice_lhs 9 10 => rw [associator_inv_naturality]
-  slice_lhs 10 12 => rw [← tensor_comp, ← tensor_comp, ← tensor_μ_def₂, tensor_comp, tensor_comp]
-  dsimp
-  coherence
+        ((X₁ ⊗ X₂) ◁ tensor_μ C (Y₁, Y₂) (Z₁, Z₂)) ≫ tensor_μ C (X₁, X₂) (Y₁ ⊗ Z₁, Y₂ ⊗ Z₂) := by
+  dsimp only [tensor_obj, prodMonoidal_tensorObj, tensor_μ]
+  simp only [whiskerRight_tensor, comp_whiskerRight, whisker_assoc, assoc, Iso.inv_hom_id_assoc,
+    tensor_whiskerLeft, braiding_tensor_left, MonoidalCategory.whiskerLeft_comp,
+    braiding_tensor_right]
+  calc
+    _ = 𝟙 _ ⊗≫
+      X₁ ◁ ((β_ X₂ Y₁).hom ▷ (Y₂ ⊗ Z₁) ≫ (Y₁ ⊗ X₂) ◁ (β_ Y₂ Z₁).hom) ▷ Z₂ ⊗≫
+        X₁ ◁ Y₁ ◁ (β_ X₂ Z₁).hom ▷ Y₂ ▷ Z₂ ⊗≫ 𝟙 _ := ?eq1
+    _ = _ := ?eq2
+  case eq1 => coherence
+  case eq2 => rw [← whisker_exchange]; coherence
```

Comment:
My preference is not to use this `?eqn` style, but to write the proofs inline in the `calc` block, at least when the proofs are as short as these ones.

I'm happy to use this technique for multiline proofs needed in a calc block, however.